README.md

```diff
@@ -153,150 +153,37 @@ If you have problems, you can review the [documentation](https://os.mbed.com/doc
 for suggestions on what could be wrong and how to fix it.
 
 ## Changing the block device
-
-In Mbed OS, a C++ classes that inherits from the [BlockDevice](https://os.mbed.com/docs/latest/reference/storage.html#block-devices)
-interface represents each block device. You can change the filesystem in the
-example by changing the class declared in main.cpp.
-
-``` diff
--SPIFBlockDevice bd(
--        MBED_CONF_SPIF_DRIVER_SPI_MOSI,
--        MBED_CONF_SPIF_DRIVER_SPI_MISO,
--        MBED_CONF_SPIF_DRIVER_SPI_CLK,
--        MBED_CONF_SPIF_DRIVER_SPI_CS);
-+SDBlockDevice bd(
-+        MBED_CONF_SD_SPI_MOSI,
-+        MBED_CONF_SD_SPI_MISO,
-+        MBED_CONF_SD_SPI_CLK,
-+        MBED_CONF_SD_SPI_CS);
+Mbed-OS supports a variety of block device types, more information on supported devices can be found [here](https://os.mbed.com/docs/mbed-os/latest/apis/data-storage.html#Default-BlockDevice-configuration).
+
+Each device is represented by a C++ class that inherits from the interface class [BlockDevice](https://os.mbed.com/docs/latest/reference/storage.html#block-devices). These classes take their default configuration from the component configuration file. This may be found in `/mbed-os/storage/blockdevice/` under the path corresponding to the block device type—for example [mbed_lib.json](https://github.com/ARMmbed/mbed-os/blob/master/storage/blockdevice/COMPONENT_SPIF/mbed_lib.json). 
+
+In this example, you can determine which block device is used by modifying the declaration of the object `bd` in main.cpp. For instance, if you wanted to add a SPIF block device you would need to declare `bd` as an object of the type SPIFBlockDevice.
+``` c++
+SPIFBlockDevice bd(
+      MBED_CONF_SPIF_DRIVER_SPI_MOSI,
+      MBED_CONF_SPIF_DRIVER_SPI_MISO,
+      MBED_CONF_SPIF_DRIVER_SPI_CLK,
+      MBED_CONF_SPIF_DRIVER_SPI_CS
+      );
 ```
+Then, you would override the default device settings in `mbed_app.json`, defining any pin reassignments here:
```

Comment:
You previsouly explained the two step process: 
1- Check whether you need to add the component
2- Possibly modify the pin assignment
I think that it was a good explanation